wgengine/wgcfg/parser.go

```diff
@@ -83,12 +91,10 @@ func FromUAPI(r io.Reader) (*Config, error) {
 		if line == "" {
 			continue
 		}
-		parts := strings.Split(line, "=")
-		if len(parts) != 2 {
-			return nil, fmt.Errorf("failed to parse line %q, found %d =-separated parts, want 2", line, len(parts))
+		key, value, ok := stringsCut(line, "=")
```

Comment:
better is to not work with strings at all. The scanner.Text() on line 90 allocates on each line.

But then line 98 goes back to working in []byte.

Ideally everything would work in terms of mem.RO (https://pkg.go.dev/go4.org/mem#RO)

Reply:
I'll try this out before merging